bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -3004,6 +3007,246 @@ private static boolean containsType(Set<Object> valueSpace, Type type) {
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side value is greater than the right hand side value, else false.
+     */
+    public static boolean compareValueGreaterThan(Object lhsValue, Object rhsValue) {
+        return compareValues(lhsValue, rhsValue, "") > 0;
+    }
+
+    /**
+     * Check if left hand side value is greater than or equal to the right hand side value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side value is greater than or equal to the right hand side value, else false.
+     */
+    public static boolean compareValueGreaterThanOrEqual(Object lhsValue, Object rhsValue) {
+        return compareValues(lhsValue, rhsValue, "") >= 0;
+    }
+
+    /**
+     * Check if left hand side float value is less than right hand side float value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is less than right hand side float value, else false.
+     */
+    public static boolean compareValueLessThan(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue, "") < 0;
+    }
+
+    /**
+     * Check if left hand side float value is less than or equal to the right hand side float value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is less than or equal to the right hand side float value, else false.
+     */
+    public static boolean compareValueLessThanOrEqual(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue, "") <= 0;
+    }
+
+    /**
+     * Check if left hand side float value is greater than the right hand side float value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is greater than the right hand side float value, else false.
+     */
+    public static boolean compareValueGreaterThan(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue, "") > 0;
+    }
+
+    /**
+     * Check if left hand side float value is greater than or equal to the right hand side float value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is greater than or equal to the right hand side float value, else
+     * false.
+     */
+    public static boolean compareValueGreaterThanOrEqual(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue, "") >= 0;
```

Comment:
without passing `""` in several places, you can have an overloaded method that takes in only `lhsValue` & `rhsValue`.